### DISCUSSION TITLE
how to get tags assigned to a file or a folder?

### DISCUSSION THREAD

**Discussion by zhongzishi** [cloud-py-api/nc_py_api#258] [Q&A]
Hi,

I passed through the code and found that it seems there is no method to get list of tags assigned to an object. There are methods such as assign_tag and unassign_tag, but I'm not able to find a list of tags already assigned to a file.

Please help.

Regards.

**Comment by bigcat88** [comment]
look like this functionality is missing. I will try to add it on a weekend(need to find with which DAV request I can get that info)

**Reply by zhongzishi** [reply]
That would be great! Many thanks !

BTW, you can use <oc:tags /> in propfind to get tags as mentionned in WebDAV doc : https://docs.nextcloud.com/server/latest/developer_manual/client_apis/WebDAV/basic.html

**Reply by bigcat88** [reply]
it does not work for me for some reason..

**Reply by zhongzishi** [reply]
I tried a nextcloud api wrapper, the get_tags() method works for me.
`nxc.get_folder({folder_path}).get_tags()`
returns 
`[Tag({'href' : '/remote.php/dav/systemtags-relations/files/2059/5', 'display_name' : 'test'}),
Tag({'href' : '/remote.php/dav/systemtags-relations/files/2059/9', 'display_name' : 'test6'})]`

https://github.com/luffah/nextcloud-API/blob/108b8f019c22377574767fe7b199bf543a4ca343/src/nextcloud/api_wrappers/systemtags.py#L221

**Reply by bigcat88** [reply]
good, I will look how it is done there. Or I can as usual debug nextcloud server 😂
